### ISSUE TITLE
storyshots renderOnlyWithOptions

### ISSUE THREAD

**Issue by luetkemj** [storybooks/storybook#3026]
### Issue details

I would like to be able to pass createNodeMock options to renderOnly in storyshots. I have refs in my stories that work with snapshotWithOptions but not renderOnly.

### Steps to reproduce

This works:
```
initStoryshots({
  test: snapshotWithOptions({
    createNodeMock: (element) => {
      if (element.type === 'div') {
        return document.createElement('div');
      }
      return null;
    },
  }),
});
```

this does not
```
initStoryshots({
  test: renderOnly({
    createNodeMock: (element) => {
      if (element.type === 'div') {
        return document.createElement('div');
      }
      return null;
    },
  }),
});
```

If there an existing way to handle mocking of refs in renderOnly docs should be improved so the solution is more transparent.

**Comment by igor-dv** [comment]
@luetkemj, do you want to PR it?

**Comment by luetkemj** [comment]
@igor-dv sure, I'll give it a go.

**Comment by Hypnosphi** [comment]
Released as [`4.0.0-alpha.9`](https://github.com/storybooks/storybook/releases/tag/v4.0.0-alpha.9)
